portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,24 +34,46 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    static final String COMMENT_TABLE_NAME = "Comment";
+    static final String COMMENT_COLUMN_NAME = "comment";
+    static final String TIMESTAMP_COLUMN_NAME = "submit_time";
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Convert comments to JSON
-    List<String> comments = new ArrayList<>();
-    comments.add("This is amazing!");
-    comments.add("10/10!");
-    comments.add("LGTM");
-    String json = convertToJsonUsingGson(comments);
+      Query query = new Query(COMMENT_TABLE_NAME).addSort(TIMESTAMP_COLUMN_NAME, SortDirection.DESCENDING);
+      PreparedQuery results = datastore.prepare(query);
+      List<String> commentsList = new ArrayList<String>();
+      List<Entity> limitedResults = results.asList(FetchOptions.Builder.withLimit(2));
 
-    // Send JSON as response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
+      for (Entity entity : limitedResults) {
+          String commentInQuery = (String) entity.getProperty(COMMENT_COLUMN_NAME);
+          commentsList.add(commentInQuery);
+      }
```

Comment:
This could be a bit more java-8y like:

List<String> commentsList= limitedResults.stream()
  .map(entity -> entity.getProperty(COMMENT_COLUMN_NAME))
  .map(String.class::cast)
  .collect(toList())

Reply:
ill make this change, but maybe we could talk about this offline?